src/test/java/com/google/devtools/build/lib/analysis/util/BuildViewForTesting.java

```diff
@@ -230,7 +230,12 @@ public AnalysisResult update(
   /** Sets the configuration. Not thread-safe. */
   public void setConfigurationForTesting(
       EventHandler eventHandler, BuildConfigurationValue configuration) {
-    skyframeBuildView.setConfiguration(eventHandler, configuration, /* maxDifferencesToShow= */ -1);
+    try {
+      skyframeBuildView.setConfiguration(eventHandler, configuration,
+          /* maxDifferencesToShow= */ -1, /* failOnAnalysisCacheDiscard */ false);
+    } catch (InvalidConfigurationException e) {
+      // Never throw during testing from here.
```

Comment:
You don't expect this condition to trigger, right? i.e. your new test in `AnalysisCachingTest` doesn't trigger this code path?

If we don't expect this line to execute, how about throwing an `UnsupportedOperationException` with the message that we don't have a use case that requires handling this yet? That could make the code more future-proof against later refactorings.